library/Core.cpp

```diff
@@ -2356,6 +2356,18 @@ bool Core::DFH_ncurses_key(int key)
     return ncurses_wgetch(key, dummy);
 }
 
+static bool getSuppressDuplicateKeyboardEvents() {
+    auto L = Lua::Core::State;
+    color_ostream_proxy out(Core::getInstance().getConsole());
+    Lua::StackUnwinder top(L);
+    bool suppress = false;
+    Lua::CallLuaModuleFunction(out, L, "dfhack", "getSuppressDuplicateKeyboardEvents", 0, 1,
```

Comment:
We could probably just access the table value directly to simplify things, unless you're thinking you might want more logic in the function in the future.

Reply:
No, the function exists purely to give it a "standard" access method from cpp. What would direct value access look like?

Reply:
I think just `DFHack::Lua::PushModulePublic(out, L, "dfhack", "SUPPRESS_DUPLICATE_KEYBOARD_EVENTS")` should be enough to get it on the stack, followed by some call to `lua_toboolean()` to retrieve it.

Reply:
done. A few other calls to Lua inside of Core.cpp could be simplified as well, either with this method (for fields) or by using `Lua::CallLuaModuleFunction` (for functions), but that can be done some other time.